### PR TITLE
Use strict dependencies in CI

### PR DESCRIPTION
**Pull request by SDxKeeper**
### Details:
Follow up for #5473
Shall speed up builds on Linux.

### CODE REVIEW

**Comment by SDxKeeper** [comment]
@asomsiko FYI

File: .ci/azure/linux_conditional_compilation.yml [hunk 1]
```diff
@@ -71,6 +71,7 @@ jobs:
         -DENABLE_FASTER_BUILD=ON
         -DENABLE_PROFILING_ITT=ON
         -DSELECTIVE_BUILD=COLLECT
+        -DENABLE_STRICT_DEPENDENCIES=ON
```
Comment (ilya-lavrenov) [inline .ci/azure/linux_conditional_compilation.yml:74]: why do we need it here? ngraph unit tests are not needed for CC

File: .ci/azure/mac.yml [hunk 2]
```diff
@@ -90,7 +90,7 @@ jobs:
       # Disable errors with Ninja
       export CXXFLAGS="-Wno-error=unused-command-line-argument"
       export CFLAGS="-Wno-error=unused-command-line-argument"
-      cmake -GNinja -DVERBOSE_BUILD=ON -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DIE_EXTRA_MODULES=$(OPENVINO_CONTRIB_REPO_DIR)/modules $(REPO_DIR)
+      cmake -GNinja -DVERBOSE_BUILD=ON -DCMAKE_BUILD_TYPE=$(BUILD_TYPE) -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DENABLE_STRICT_DEPENDENCIES=ON -DIE_EXTRA_MODULES=$(OPENVINO_CONTRIB_REPO_DIR)/modules $(REPO_DIR)
```
Comment (ilya-lavrenov) [inline .ci/azure/mac.yml:93]: as far as I understand, `ENABLE_STRICT_DEPENDENCIES` are needed for developers only (so it must be `ON` by default), while on CI we must allow build to be parallelized as much as possible

Comment (SDxKeeper) [reply .ci/azure/mac.yml:93]: unfortunately it's OFF by default, see changes in  #5473

**Pull request closed without merging**